boa/src/builtins/value/operations.rs

```diff
@@ -170,6 +136,25 @@ impl Not for Value {
     }
 }
 
+/// The internal comparison abstract operation SameValueZero(x, y),
+/// where x and y are ECMAScript language values, produces true or false.
+/// SameValueZero differs from SameValue only in its treatment of +0 and -0.
+///
+/// Such a comparison is performed as follows:
+///
+/// https://tc39.es/ecma262/#sec-samevaluezero
+pub fn same_value_zero(x: &Value, y: &Value) -> bool {
+    if x.get_type() != y.get_type() {
+        return false;
+    }
+
+    if x.is_number() {
+        return number::same_value_zero(x, y);
+    }
+
+    same_value_non_number(x, y)
+}
+
```

Comment:
This will potentially cause issues with the same-named function in `number`. Maybe it would make sense for `number::same_value_zero` to only accept numbers?